### DISCUSSION TITLE
workflow query

### DISCUSSION THREAD

**Discussion by imethod** [Vector35/binaryninja-api#4837] [Analysis] [answered]
I used the relevant workflow python api in my project but I found that new workflow would only call my action during the first analysis 

When I reanalyzed the function ,I did not see the relevant log when the analysis was triggered

It seems ninja found out for efficiency that no changes were made to the binary and workflow was not triggered

![image](https://github.com/Vector35/binaryninja-api/assets/109936536/ee86d729-cf57-4eca-85ca-bf3fdb88281f)
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/5a22ccbf-01cd-40d0-8955-c35b846baa95)

The binary file is opened for the first time
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/306c04ca-693e-4c19-bc1b-778a11389229)

**Comment by bpotchik** [comment, marked as answer]
I tried the above code snippet and everything works as it should. The Activity was invoked on reanalysis. Are there any other details or steps that are missing? Is there a chance the Activity is getting garbage collected? You could try saving the Activity as a local object.

**Reply by imethod** [reply]
I enter this code in the python console ,Then switch workflow in Settings,Finally Analysis ->Reanalyze
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/08e943eb-3697-4083-b4ac-54cf0ae4c14a)
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/1f2afcef-01f5-4c07-b4a7-bf82cb10d3d3)
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/f2fa5672-5431-4e8d-be13-e29686fc226f)
Did not see the related log

**Reply by psifertex** [reply]
I'm going to mark this one as answered as I believe Brian's original guess above is likely correct. If you put the following file into your plugin folder and use that you should see it's triggered each time you reanalyze as intended. It's very likely the python console is garbage collecting your objects and that's why you're not seeing it triggered.
[pwf.py.zip](https://github.com/Vector35/binaryninja-api/files/13956470/pwf.py.zip)

**Reply by imethod** [reply]
Yes, success.
